functorch/examples/maml_omniglot/maml-omniglot-transforms.py

```diff
@@ -104,42 +105,38 @@ def main():
 
     net.train()
 
-    # Given this module we've created, rip out the parameters and buffers
-    # and return a functional version of the module. `fnet` is stateless
-    # and can be called with `fnet(params, buffers, args, kwargs)`
-    fnet, params, buffers = make_functional_with_buffers(net)
-
     # We will use Adam to (meta-)optimize the initial parameters
     # to be adapted.
-    meta_opt = optim.Adam(params, lr=1e-3)
+    meta_opt = optim.Adam(net.parameters(), lr=1e-3)
 
     log = []
     for epoch in range(100):
-        train(db, [params, buffers, fnet], device, meta_opt, epoch, log)
-        test(db, [params, buffers, fnet], device, epoch, log)
+        train(db, net, device, meta_opt, epoch, log)
+        test(db, net, device, epoch, log)
         plot(log)
 
 
 # Trains a model for n_inner_iter using the support and returns a loss
 # using the query.
 def loss_for_task(net, n_inner_iter, x_spt, y_spt, x_qry, y_qry):
-    params, buffers, fnet = net
+    params = {k: v for k, v in net.named_parameters()}
+    buffers = {k: v for k, v in net.named_buffers()}
     querysz = x_qry.size(0)
 
     def compute_loss(new_params, buffers, x, y):
-        logits = fnet(new_params, buffers, x)
+        logits = functional_call(net, {**new_params, **buffers}, (x,))
         loss = F.cross_entropy(logits, y)
         return loss
 
     new_params = params
     for _ in range(n_inner_iter):
         grads = grad(compute_loss)(new_params, buffers, x_spt, y_spt)
-        new_params = [p - g * 1e-1 for p, g, in zip(new_params, grads)]
+        new_params = {k: new_params[k] - g * 1e-1 for k, g, in grads.items()}
 
     # The final set of adapted parameters will induce some
     # final loss and accuracy on the query dataset.
     # These will be used to update the model's meta-parameters.
-    qry_logits = fnet(new_params, buffers, x_qry)
+    qry_logits = functional_call(net, {**new_params, **buffers}, (x_qry,))
```

Comment:
I have some UX nits with the functional_call API:
1. Most users are not going to know that `{**new_params, **buffers}` smashes the dicts together.
2. `functional_call` accepts args and kwargs as a tuple and a dictionary, respectively.

(2) makes functional_call feel a bit bureaucratic. 

We have an opportunity to repurpose functional_call the way we want it. In particular, we are likely going to namespace functional_call into torch.func or whatever we pick. Our version of functional_call can have API improvements to it while we keep around torch.nn.utils.stateless.functional_call for BC.

Here's an idea of what a better functional call could look like:
```
# functional_call(module, dict1, dict2) returns a new callable
functional_call(net, params, buffers)(x_qry)
```

Reply:
Do you have a sense of if users have been OK with functional_call accepting args/kwargs as a tuple and dictionary?

Reply:
I did a quick search of issues and Google and didn't see anything pro or against it, but I think one thing to note is that most forward calls just take in a single tensor of inputs. So, I think it's somewhat telling that functional_call now supports just taking in a single tensor. Specifically, I would guess that people kept passing in a single tensor instead of a tuple of a single tensor and got annoyed at either the error message or silent correctness issue